### ISSUE TITLE
Optional boolean as query parameter

### ISSUE THREAD

**Issue by allenpsuner** [domaindrivendev/Swashbuckle#339]
I have an optional boolean query param in a GetAll Models function in this format:

```
        [ResponseType(typeof(IEnumerable<Foo>))]
        public IEnumerable<Foo> GetAll(bool? myBool = null)
        {
            if (myBool.HasValue)
            {
                return Repository.GetAll(Url).Where<Foo>(foo => foo.myBoolValue.Equals(myBool));
            }
            else
            {
                return Repository.GetAll(Url);
            }
        }
```

But in the swagger UI it doesn't behave like a true nullable.

![screen shot 2015-05-19 at 3 40 26 pm](https://cloud.githubusercontent.com/assets/10819359/7715554/78d5515c-fe3d-11e4-890f-c37fe62e296e.png)

Is this a bug that is being worked on? Or is there a possible workaround that I should be using?

**Comment by monobjorn** [comment]
Maybe this has to do with Swagger UI? Have a look at: [non-required Boolean value is always selected #1191] (https://github.com/swagger-api/swagger-ui/issues/1191)

**Comment by allenpsuner** [comment]
Maybe I should give some context. I'm using *Swashbuckle* because I heard that it was super easy to integrate into my .NET WebApi project. I'm pretty sure there are no YAML configs for Swashbuckle, so that thread doesn't really help me out. I'm 99% sure that this isn't a bug and I'm just doing something wrong, but Stack-Overflow wasn't being responsive quick enough.

**Comment by monobjorn** [comment]
I'm also new to Swashbuckle (just started using it the day before yesterday).  
Swashbuckle generates a JSON specification of your service. This corresponds to the YAML mentioned in the issue.

I tried adding an optional boolean to my project and I got the same issue that you had. I then downloaded the newest version of [swagger-ui](https://github.com/swagger-api/swagger-ui) and then it worked. So, it's not actually a Swashbuckle-issue.

**Comment by domaindrivendev** [comment]
@monobjorn is absolutely correct - this was a swagger-ui issue as opposed to a bug in SB itself.

I've upgraded to the latest swagger-ui and the issue is fixed. It will be available with the next release which I'm working hard to get out ASAP.

**Comment by allenpsuner** [comment]
Awesome. Thank you so much @domaindrivendev.

**Comment by domaindrivendev** [comment]
Should be resolved in latest release - 5.2.0
